internal/jimm/serviceaccount/serviceaccount.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2025 Canonical.
+
+package serviceaccount
+
+import (
+	"context"
+
+	jujuparams "github.com/juju/juju/rpc/params"
+	"github.com/juju/names/v5"
+
+	"github.com/canonical/jimm/v3/internal/db"
+	"github.com/canonical/jimm/v3/internal/errors"
+	"github.com/canonical/jimm/v3/internal/openfga"
+	ofganames "github.com/canonical/jimm/v3/internal/openfga/names"
+	jimmnames "github.com/canonical/jimm/v3/pkg/names"
+)
+
+type CredentialCopier interface {
+	CopyCredential(ctx context.Context, originalUser *openfga.User, newUser *openfga.User, cred names.CloudCredentialTag) (names.CloudCredentialTag, []jujuparams.UpdateCredentialModelResult, error)
+}
+
+// serviceAccountManager provides a means to manage service
+// accounts within JIMM.
+type serviceAccountManager struct {
+	store      *db.Database
+	authSvc    *openfga.OFGAClient
+	credCopier CredentialCopier
+}
+
+// NewServiceAccountManager returns a new serviceAccountManager that
+// provides methods to manage service accounts.
+func NewServiceAccountManager(store *db.Database, authSvc *openfga.OFGAClient, cc CredentialCopier) (*serviceAccountManager, error) {
+	if store == nil {
+		return nil, errors.E("role store cannot be nil")
+	}
+	if authSvc == nil {
+		return nil, errors.E("role authorisation service cannot be nil")
+	}
+	if cc == nil {
+		return nil, errors.E("credential copier service cannot be nil")
+	}
+	return &serviceAccountManager{store, authSvc, cc}, nil
+}
+
+// AddServiceAccount checks that no one owns the service account yet
```

Comment:
ah, ok. this is a bit confusing and took me 2min to wrap my thoughts around it: when adding a service account, we check that nobody else has administrator relation to it - assert that there is only one administrator relation to it. but once the account is added, the administrator is allowed to add further administrators to the service account, right?

e.g. i create a service account and then allow a group of users to administer it (by adding their cloud credentials).